app/main.py

```diff
@@ -1,34 +1,69 @@
+from typing import List, Tuple
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(
+            self,
+            row: int,
+            column: int,
+            is_alive: bool = True
+    ) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: Tuple[int, int],
+            end: Tuple[int, int],
+            is_drowned: bool = False
+    ) -> None:
+        self.is_drowned = is_drowned
+        self.decks = [Deck(row, column) for row in range(start[0], end[0] + 1)
+                      for column in range(start[1], end[1] + 1)]
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def get_deck(
+            self,
+            row: int,
+            column: int
+    ) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(self, row: int, column: int) -> str:
+        if not self.is_drowned:
+            deck = self.get_deck(row, column)
+            if deck:
+                deck.is_alive = False
+                if all(not d.is_alive for d in self.decks):
```

Comment:
Avoid using single-letter variable names like `d`; use a more descriptive name such as `deck` for better readability.